src/lib/Microsoft.Health.Events/EventHubProcessor/EventProcessor.cs

```diff
@@ -29,12 +29,29 @@ public override async Task RunAsync(CancellationToken ct)
             EventProcessorClient.ProcessErrorAsync += ProcessErrorHandler;
             EventProcessorClient.PartitionInitializingAsync += ProcessInitializingHandler;
 
-            try
+            // Try to connect and read events
+            bool connected = false;
+            while (!connected)
```

Comment:
This looks correct to me. The other option is to implement a Polly retry policy here. Not sure if that is overkill